charforge-gui/backend/app/api/training.py

```
@@ -0,0 +1,295 @@
+        session.status = "completed" if result["success"] else "failed"
+        session.completed_at = datetime.utcnow()
+        session.progress = 100.0 if result["success"] else session.progress
+        
+        # Update character status
+        character.status = "completed" if result["success"] else "failed"
+        if result["success"]:
+            character.completed_at = datetime.utcnow()
+        
+        db.commit()
+        
+    except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)